### DISCUSSION TITLE
How do you remove unreachable songs from the DB?

### DISCUSSION THREAD

**Discussion by SuperFola** [ampache/ampache#3488] [Q&A] [answered]
I had a weird setup and ended up adding music twice. First time from /music/folder/folder/song.mp3, second time from /music/folder/song.mp3. The song has been moved from the first path to the second, and now when I search for "song" I have duplicates, but only one plays because its path is correct.

How do you clean the DB of such unreachable songs? Cleaning the catalog doesn't seem to get rid of those. I'm fine with having to do some SQL

**Comment by lachlan-00** [comment, marked as answer]
in cli you can clean https://github.com/ampache/ampache/wiki/cli-faq

```
php bin/cli run:updateCatalog music -c
```

help will also work on any commands
```
bin/cli --help
bin/cli run:updateCatalog --help
```

**Reply by lachlan-00** [reply]
i'll need to expand this wiki page a bit more and put links in the cli

**Reply by SuperFola** [reply]
Thanks, I must have missed it!

Something weird however, it tells me it deleted 0 files, but when I try to play my duplicate songs, still the same as before: one works (correct path), the other doesn't (bad path)

![](https://user-images.githubusercontent.com/10694085/226742933-700a9c82-b3c3-4ecb-9b35-accaef1bfce0.png)

**Reply by lachlan-00** [reply]
can you double check the file path of each file? it's likely to be a duplicate but different files

**Reply by SuperFola** [reply]
Yes, they are duplicates with different paths, with a path being reachable, while the other isn't (I put music in the wrong folder, updated the catalog, noticed the error and moved the files, then updated again).

**Reply by lachlan-00** [reply]
it might be an error reading the path. both files are part of a valid catalog? 

e.g my music catalog is /mnt/music/

* /mnt/music/album/file1.mp3
* /mnt/music/album/file2.mp3

If your file is not part of a catalog path like /mnt/notmusic/album/file1.mp3 the check process might not like that as most of our checks are based on the catalog path.

you can also use the cli to check the file individually too.

```
php bin/cli run:updateCatalogFile music-catalog /mnt/music/album/file1.mp3
```

You can also just enable delete_from_disk in the config file and an X will show up on song rows to delete them but it would be good to work out how it's not working.

turn on logging and let me know what it tells you if you can check that path using hte cli

**Reply by SuperFola** [reply]
Yes, both files are part of my /mnt/Music catalog, only difference between them is that one of them points to a non-existing file.

I'll update the catalog and get some logs, thanks for your time!

**Reply by SuperFola** [reply]
Trying to play the "bad path" song yields this (among other lines about the db connection booting up, web player starting, etc)

```
2023-03-25T08:58:07+00:00 [user] (play/index) -> Media /media/test Music/test Music/Muse - Absolution/12 - Endlessly.mp3 (Endlessly) does not have a valid filename specified
```

The correct path should be `/media/test Music/Muse - Absolution/12 - Endlessly.mp3 (Endlessly)`

**Comment by lachlan-00** [comment]
I think there was a bug in smaller catalogs as well that's been fixed (if you use develop branch)

**Reply by SuperFola** [reply]
I updated the container to the latest version. After a clean, garbage collect (not sure what this does) verify and update the unreachable files are still there.

That may just be a problem with my data, do you think I can purge the catalog data and keep the playlists somewhere to import them again? That might be the easiest thing to do

**Comment by SuperFola** [comment]
This has been solved by deleting the database and setting everything up again, I just had to recreate my playlists. I may have missed it, is there a way to export playlists to m4u format (or something else based on the files paths?)

**Reply by lachlan-00** [reply]
you can export playlists using the cli commands to export as local files.

basic command is
```
php bin/cli export:playlist ~/playlists/
```

you can add a --help to get command info. the m3u lists should export the file paths as utf8
